source/common/stats/thread_local_store.cc

```diff
@@ -500,8 +501,12 @@ Histogram& ThreadLocalStoreImpl::ScopeImpl::histogramFromStatName(StatName name,
   } else {
     TagExtraction extraction(parent_, final_stat_name);
 
+    // Combine the provided tags with the extracted tags.
+    std::vector<Tag> allTags = tags;
+    allTags.insert(allTags.end(), extraction.tags().begin(), extraction.tags().end());
+
     RefcountPtr<ParentHistogramImpl> stat(new ParentHistogramImpl(
-        final_stat_name, unit, parent_, *this, extraction.tagExtractedName(), extraction.tags()));
+        final_stat_name, unit, parent_, *this, extraction.tagExtractedName(), tags));
```

Comment:
You need the central & TLS caches to look at the tags as part of their key, if they are not part of the passed-in StatName.